CMakeLists.txt

```diff
@@ -20,6 +20,29 @@ endif ()
 project(remill)
 cmake_minimum_required(VERSION 3.2)
 
+if(EMSCRIPTEN)
+  set(CMAKE_BUILD_TYPE Release)
+  set(gflags_DIR "$ENV{REPOS}/gflags/build")
+  set(glog_DIR "$ENV{REPOS}/glog/build")
+  set(LLVM_DIR "$ENV{REPOS}/llvm-project/llvm/build/lib/cmake/llvm/")
+  set(remill_DIR "$ENV{REPOS}/remill/build/")
+
+  # We want to manually invoke main for Lift instead of Emscripten calling it for us
+  # So we disable INVOKE_RUN, export callMain (and FS for files).
+  set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/web/src")
+  set(CMAKE_CXX_FLAGS "\
+    ${CMAKE_CXX_FLAGS} \
+    $ENV{EM_CXX_FLAGS} \
+    -D__i386__ \
```

Comment:
Maybe you want `-m32` here? Also, probably `-mno-sse`, `-mno-avx`, and perhaps `-freestanding`?

Reply:
With `-ffreestanding` it breaks the closure compiler (a Java tool that Emscripten runs to minify and elide JS functions). I have a feeling this option is untested under Emscripten.

As for the rest, with those specified it does compile, although I don't truly know if they have any effect. Emscripten for the longest time already had SSE disabled (and probably AVX), which I think is up-streamed into clang for the WASM backend. Let me know if you'd like me to check this change in anyways.